sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -73,17 +73,7 @@ class JDBCOptions(
   // ------------------------------------------------------------
   // Optional parameters
   // ------------------------------------------------------------
-  val driverClass = {
-    val userSpecifiedDriverClass = parameters.get(JDBC_DRIVER_CLASS)
-    userSpecifiedDriverClass.foreach(DriverRegistry.register)
-
-    // Performing this part of the logic on the driver guards against the corner-case where the
-    // driver returned for a URL is different on the driver and executors due to classpath
-    // differences.
-    userSpecifiedDriverClass.getOrElse {
-      DriverManager.getDriver(url).getClass.getCanonicalName
-    }
-  }
+  val driverClass = parameters.get(JDBC_DRIVER_CLASS)
```

Comment:
Might be too late, but should these be private?
Does it solve the issue to make this a def instead?